File: 2678/NumberOfSeniorCitizens.cpp
/*
https://leetcode.com/problems/number-of-senior-citizens/description/?envType=daily-question&envId=2024-08-01
*/

class Solution {
public:
    int countSeniors(vector<string>& details) {
        int res=0;
        for(auto &vals: details){
            if(vals[11]-'0'>=6){
                if(vals[12]-'0'==0){
                    if(vals[11]-'0'!=6) res++;
                } else res++;
            }
        }
        return res;
    }
};

File: 2/2000s/26s/2678/NumberOfSeniorCitizens.js
/*
https://leetcode.com/problems/number-of-senior-citizens/description/?envType=daily-question&envId=2024-08-01
*/

/**
 * @param {string[]} details
 * @return {number}
 */
var countSeniors = function(details) {
    let resultantCitizens=0;
    let lengthOfDetails=details.length;
    for(let i=0; i<lengthOfDetails; ++i){
        let currentCitizenDetails=details[i];
        let onesDigitInAge=parseInt(currentCitizenDetails[11], 10), zerothDigitInAge=parseInt(currentCitizenDetails[12], 10);
        if(onesDigitInAge>=6){
            if(zerothDigitInAge===0){
                if(onesDigitInAge!==6) resultantCitizens++;
            } else resultantCitizens++; 
        }
    }
    return resultantCitizens;
};